src/reanimated2/animation/decay.ts

```diff
@@ -59,40 +62,81 @@ export function withDecay(
     const VELOCITY_EPS = Platform.OS !== 'web' ? 1 : 1 / 20;
     const SLOPE_FACTOR = 0.1;
 
-    function decay(animation: InnerDecayAnimation, now: number): boolean {
-      const {
-        lastTimestamp,
-        startTimestamp,
-        initialVelocity,
-        current,
-        velocity,
-      } = animation;
-
-      const deltaTime = Math.min(now - lastTimestamp, 64);
-      const v =
-        velocity *
-        Math.exp(
-          -(1 - config.deceleration) * (now - startTimestamp) * SLOPE_FACTOR
-        );
-      animation.current =
-        current + (v * config.velocityFactor * deltaTime) / 1000; // /1000 because time is in ms not in s
-      animation.velocity = v;
-      animation.lastTimestamp = now;
+    let decay: (animation: InnerDecayAnimation, now: number) => boolean;
 
-      if (config.clamp) {
-        if (initialVelocity < 0 && animation.current <= config.clamp[0]) {
-          animation.current = config.clamp[0];
-          return true;
-        } else if (
-          initialVelocity > 0 &&
-          animation.current >= config.clamp[1]
-        ) {
-          animation.current = config.clamp[1];
+    if (config.softClamp) {
+      decay = (animation: InnerDecayAnimation, now: number): boolean => {
+        const {
+          lastTimestamp,
+          startTimestamp,
+          current,
+          initialVelocity,
+          velocity,
+        } = animation;
+
+        const deltaTime = Math.min(now - lastTimestamp, 64);
+        const clampIndex = initialVelocity > 0 ? 1 : 0;
+        let deriveration = 0;
```

Comment:
What is "deriveration"? Maybe use simply something like "distance" or you wanted to use "deviation"?

Reply:
Or "derivative"?